Tests/ConvertFrom-BMOtterScriptExpression.Tests.ps1

```diff
@@ -156,8 +156,31 @@ Describe 'ConvertFrom-BMOtterScriptExpression' {
         GivenValue '@(@(), @())'
         WhenConverting
         ThenEqual @(@(), @())
+
         GivenValue '@(%(), %())'
         WhenConverting
         ThenEqual @(@{}, @{})
+
+        GivenValue ''
+        WhenConverting
+        ThenEqual ''
+    }
+
+    It 'should remove empty items from vectors' {
+        GivenValue '@(one, )'
+        WhenConverting
+        ThenEqual @('one')

```

Comment:
According to the rules of OtterScript, shouldn't the missing value be converted to an empty string? E.g., `@('one', '')`

Reply:
I got this behavior from the BuildMaster UI, when it saves a vector variable it strips all _empty_ vector items.

But in looking at the OtterScript [Formal Grammar](https://docs.inedo.com/docs/otterscript-formal-grammar#conventions) you're right, an empty string value should be supported. An empty string is a valid literal expression and a vector expression can contain any number of literal expressions.

So the OtterScript formal grammar and the BuildMaster UI do not agree. I'll ask Inedo about this.

Reply:
Looking at the OtterScript [Strings & Values](https://docs.inedo.com/docs/executionengine-otterscript-strings-and-literals) docs again, I noticed the following from the "Implicit" string syntax (emphasis my own):

> any characters except newlines and the expected delimiter (which can be a comma, right-parenthesis, right-brance, or semi-colon, depending on context); ***surrounding whitespace is always trimmed***

So what's happening in the UI is that when I submit the value `@(one, )`, it's trimming the whitespace around the second "empty" item which leads to a nil item rather than an empty string item. In order to create the vector with an empty string item, I have to use `@(one, '')`, which makes sense.

I'll get the function fixed up to support creating empty string items in vectors.